worldedit-core/src/main/java/com/sk89q/worldedit/EditSession.java

```diff
@@ -1729,22 +1813,30 @@ public int drainArea(BlockVector3 origin, double radius, boolean waterlogged, bo
         checkNotNull(origin);
         checkArgument(radius >= 0, "radius >= 0 required");
 
+        //FAWE start - liquidmask
         Mask liquidMask;
         if (plants) {
-            liquidMask = new BlockTypeMask(this, BlockTypes.LAVA, BlockTypes.WATER,
-                                           BlockTypes.KELP_PLANT, BlockTypes.KELP, BlockTypes.SEAGRASS, BlockTypes.TALL_SEAGRASS);
+            liquidMask = new BlockMaskBuilder().addTypes(BlockTypes.LAVA, BlockTypes.WATER,
+                    BlockTypes.KELP_PLANT, BlockTypes.KELP, BlockTypes.SEAGRASS, BlockTypes.TALL_SEAGRASS).build(extent);
```

Comment:
this feels unnecessary. Literally using a constructor through a builder and then even changing the variables given (this -> extent)

Reply:
Fawe discourages BlockTypeMasks in favor of our BlockMaskBuilder.